### PR TITLE
refactor(react-ui): factor out shell provider

### PR DESCRIPTION
**Pull request by wittjosiah**

### CODE REVIEW

**Comment by wittjosiah** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2581** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2581" target="_blank">Graphite</a> 
    * **PR #2592** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2592" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/2592?utm_source=stack-comment).

File: packages/apps/patterns/react-ui/src/composites/Shell/ShellContext.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  useEffect,
+  useMemo,
+  useState
+} from 'react';
+
+import {
+  IFrameClientServicesProxy,
+  MemoryShellRuntime,
+  type PublicKey,
+  type ShellController,
```
Comment (thure) [inline packages/apps/patterns/react-ui/src/composites/Shell/ShellContext.tsx:20]: Ooh, I didn’t realize this syntax was available. Nice.

**Review by thure — APPROVED** [review]
lgtm!

**Comment by wittjosiah** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: nxRgruxohA5dORPwjnRK.
See details on [graphite.dev](https://app.graphite.dev/github/pr/dxos/dxos/2592)

**Pull request merged**